libs/wingc/src/lsp/hover.rs

```diff
@@ -76,7 +259,7 @@ pub fn on_hover<'a>(params: lsp_types::HoverParams) -> Option<Hover> {
 
```

Comment:
Maybe we can extract some of this into separate functions just to make the code a bit more self-documenting / reduce the amount of nesting. Here's a sketch of one way it could look:
```rust
if let Ok(symbol_lookup) = symbol_lookup {
  format_symbol_with_type_info(symbol_name, symbol_kind)
} else if let Some(reference) = reference {
  format_reference_symbol(symbol_name, reference)
} else {
  format_unknown_symbol(symbol_name)
}
```

Reply:
Agreed, I went over this whole part of the code again and I think it's a lot better now.